src/components/ComboBox/ComboBox.jsx

```diff
@@ -0,0 +1,218 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Tag } from 'carbon-components-react';
+
+import { settings } from '../../constants/Settings';
+
+import CarbonComboBox from './CarbonComboBox';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  // eslint-disable-next-line react/forbid-foreign-prop-types
+  ...CarbonComboBox.propTypes,
+  loading: PropTypes.bool,
+  // Callback that is called with the value of the input on change
+  onChange: PropTypes.func.isRequired,
+  // OPtional classname to be applied to wrapper
+  wrapperClassName: PropTypes.string,
+  // String to pass to input field option
+  editOptionText: PropTypes.string,
+  // String to pass for tags close button aria-label. Will be prepended to value name
+  closeButtonText: PropTypes.string,
+  // Bit that will allow mult value and tag feature
+  hasMultiValue: PropTypes.bool,
+};
+
+const defaultProps = {
+  ...CarbonComboBox.defaultProps,
+  loading: false,
+  inline: false,
+  wrapperClassName: null,
+  closeButtonText: 'Close',
+  editOptionText: '-Create',
+  hasMultiValue: false,
+  items: [],
+};
+
+const ComboBox = ({
+  inline,
+  loading,
+  wrapperClassName,
+  closeButtonText,
+  editOptionText,
+  hasMultiValue,
+  itemToString,
+  onChange,
+  items,
+  downshiftProps,
+  ...comboProps
+}) => {
+  // Ref for the combobox input
+  const comboRef = React.createRef();
+  // Input value that is added to list
+  const [inputValue, setInputValue] = useState('');
+  // Current selected item that shows in the input
+  const [selectedItem, setSelectedItem] = useState(null);
+  // Array that populates list
+  const [listItems, setListItems] = useState(items);
+  // Array that populates tags
+  const [tagItems, setTagItems] = useState([]);
+  // Highlighted index for list dropdown
+  const [highlightedIndex, setHighlightedIndex] = useState(-1);
+
+  useEffect(() => {
+    // If there are tags then clear and focus the input
+    if (hasMultiValue) {
+      setSelectedItem(null);
+      // comboRef.current.textInput.current.focus();
+    }
+  });
+
+  const handleOnClose = e => {
+    // Get close target's text
+    const closedValue = e.currentTarget.parentNode.children[0].textContent;
```

Comment:
Does there need to be any null checks here? I'm not sure how likely it would be that the textContent doesn't exist?

Reply:
Unlikely that most of these will be nulls but will put some on the children. in case of a markup change.